### ISSUE TITLE
Edge Agent's storage directory is persisted on the host filesystem - Warning

### ISSUE THREAD

**Issue by ThomasPe** [Azure/iotedge#3305]
## Expected Behavior
I'm trying to fix the Warning that is output by `sudo iotedge check`

```
‼ production readiness: Edge Agent's storage directory is persisted on the host filesystem - Warning
    The edgeAgent module is not configured to persist its /tmp/edgeAgent directory on the host filesystem.
    Data might be lost if the module is deleted or updated.
    Please see https://aka.ms/iotedge-storage-host for best practices.
```
I've set up the bindings in my deployment file like this 
```
          "edgeAgent": {
            "type": "docker",
            "settings": {
              "image": "mcr.microsoft.com/azureiotedge-agent:1.0",
              "createOptions": {
                "HostConfig": {
                  "Binds": [
                    "/etc/iotedge/storage/:/iotedge/storage/"
                  ]
                }
              }
            },
            "env": {
              "storageFolder": {
                "value": "/iotedge/storage/"
              }
            }
          }
```

This config for edgeHub worked just fine and I'm not receiving any more errors for it:
```
          "edgeHub": {
            "type": "docker",
            "status": "running",
            "restartPolicy": "always",
            "settings": {
              "image": "mcr.microsoft.com/azureiotedge-hub:1.0",
              "createOptions": {
                "HostConfig": {
                  "Binds": [
                    "/etc/iotedge/storage/:/iotedge/storage/"
                  ],
                  "PortBindings": {
                    "5671/tcp": [
                      {
                        "HostPort": "5671"
                      }
                    ],
                    "8883/tcp": [
                      {
                        "HostPort": "8883"
                      }
                    ],
                    "443/tcp": [
                      {
                        "HostPort": "443"
                      }
                    ]
                  }
                }
              }
            },
            "env": {
              "storageFolder": {
                "value": "/iotedge/storage/"
              }
            }
          }
```

How do I get the agent to pick up these settings? I've restarted the runtime, the PC, the docker container - nothing worked.
Again, same settings for edgeHub worked perfectly. Am I missing something here?

## Current Behavior
Still getting the same warning from `iotedge check`.

## Steps to Reproduce
Provide a detailed set of steps to reproduce the bug.
1. change deployment template to include bindings
2. create deployment manifest
3. deploy
4. run `sudo iotedge check`

### Output of `iotedge check`

<details>
<summary>Click here</summary>

```

√ config.yaml is well-formed - OK
√ config.yaml has well-formed connection string - OK
√ container engine is installed and functional - OK
‼ config.yaml has correct hostname - Warning
    config.yaml has hostname 8P0WN42 which does not comply with RFC 1035.

    - Hostname must be between 1 and 255 octets inclusive.
    - Each label in the hostname (component separated by ".") must be between 1 and 63 octets inclusive.
    - Each label must start with an ASCII alphabet character (a-z, A-Z), end with an ASCII alphanumeric character (a-z, A-Z, 0-9), and must contain only ASCII alphanumeric characters or hyphens (a-z, A-Z, 0-9, "-").

    Not complying with RFC 1035 may cause errors during the TLS handshake with modules and downstream devices.
√ config.yaml has correct URIs for daemon mgmt endpoint - OK
√ latest security daemon - OK
√ host time is close to real time - OK
√ container time is close to host time - OK
‼ DNS server - Warning
    Container engine is not configured with DNS server setting, which may impact connectivity to IoT Hub.
    Please see https://aka.ms/iotedge-prod-checklist-dns for best practices.
    You can ignore this warning if you are setting DNS server per module in the Edge deployment.
‼ production readiness: certificates - Warning
    The Edge device is using self-signed automatically-generated development certificates.
    They will expire in 89 days (at 2020-10-26 07:34:33 UTC) causing module-to-module and downstream device communication to fail on an active deployment.
    After the certs have expired, restarting the IoT Edge daemon will trigger it to generate new development certs.
    Please consider using production certificates instead. See https://aka.ms/iotedge-prod-checklist-certs for best practices.

‼ production readiness: container engine - Warning

    Device is not using a production-supported container engine (moby-engine).
    Please see https://aka.ms/iotedge-prod-checklist-moby for details.
√ production readiness: logs policy - OK
‼ production readiness: Edge Agent's storage directory is persisted on the host filesystem - Warning
    The edgeAgent module is not configured to persist its /tmp/edgeAgent directory on the host filesystem.
    Data might be lost if the module is deleted or updated.
    Please see https://aka.ms/iotedge-storage-host for best practices.
√ production readiness: Edge Hub's storage directory is persisted on the host filesystem - OK

Connectivity checks
-------------------
√ host can connect to and perform TLS handshake with IoT Hub AMQP port - OK
√ host can connect to and perform TLS handshake with IoT Hub HTTPS / WebSockets port - OK
√ host can connect to and perform TLS handshake with IoT Hub MQTT port - OK
√ container on the default network can connect to IoT Hub AMQP port - OK
√ container on the default network can connect to IoT Hub HTTPS / WebSockets port - OK
√ container on the default network can connect to IoT Hub MQTT port - OK
√ container on the IoT Edge module network can connect to IoT Hub AMQP port - OK
√ container on the IoT Edge module network can connect to IoT Hub HTTPS / WebSockets port - OK
√ container on the IoT Edge module network can connect to IoT Hub MQTT port - OK

18 check(s) succeeded.
5 check(s) raised warnings. Re-run with --verbose for more details.
```
</details>

### Device Information
* Host OS [e.g. Ubuntu 16.04, Ubuntu 18.04, Windows IoT Core]:  Ubunutu 18.0.4
* Architecture [e.g. amd64, arm32, arm64]: amd64
* Container OS [e.g. Linux containers, Windows containers]: Linux

### Runtime Versions
* iotedged [run `iotedge version`]: 1.0.9.4
* Edge Agent [image tag (e.g. 1.0.0)]: 1.0.0
* Edge Hub [image tag (e.g. 1.0.0)]:  1.0.0
* Docker/Moby [run `docker version`]: 

Note: when using Windows containers on Windows, run `docker -H npipe:////./pipe/iotedge_moby_engine version` instead

## Logs

<details>
<summary>iotedged logs</summary>

```

<Paste here between the triple backticks>

```
</details>

<details>
<summary>edge-agent logs</summary>

```

<Paste here between the triple backticks>

```
</details>

<details>
<summary>edge-hub logs</summary>

```

<Paste here between the triple backticks>

```
</details>

## Additional Information
Please provide any additional information that may be helpful in understanding the issue.

**Comment by arsing** [comment]
If you only configured it in the Edge Agent twin and not also in the Edge Agent module spec in `config.yaml`, then it is likely that your Edge Agent module does not actually have that mount. Run `docker inspect` to confirm.

The reason is a bug that was fixed in https://github.com/Azure/iotedge/pull/2889 . Without that fix, Edge Agent is not redeployed from its initial deployment as configured in `config.yaml` to match the twin from IoT Hub if the image spec in both is the same, even if the other properties are different. So if the module spec in the `config.yaml` didn't have that mount, then the container will not have that mount, and it'll not be redeployed with the spec from the twin that does have that mount.

**Comment by ThomasPe** [comment]
You're right, the mount is missing from docker inspect. I take it the fix you mentioned is not yet part of the the latest release?

**Comment by arsing** [comment]
The fix in 1.0.10, which is currently in RC.
